packages/x-utils/src/object.ts

```diff
@@ -94,6 +94,32 @@ export function cleanUndefined<T>(obj: T): T {
       );
 }
 
+/**
+ * Creates an object picking only the properties whose values are not:
+ * - `undefined`.
+ * - `null`.
+ * - an empty string.
+ * - an empty array.
+ *
+ * @param obj - The object from whom pick the values.
+ * @returns A new object with the not empty properties of the source object.
+ * @public
+ */
+export function cleanEmpty<T extends Record<string, unknown>>(obj: T): T {
```

Comment:
I know this file is completely wrong in terms of generic naming, but given our contribution guide:

> - Generic values must have a meaningful name.

```ts
// ❌ Wrong
type Dictionary<T> = Record<string, T>;

// ✅ Good
type Dictionary<Value> = Record<string, Value>;
```

Reply:
Also, should we type this function better? At least to say that it won't return null/undefined values:

```ts
export type ExcludeEmptyProperties<Something> = {
  [Key in keyof Something as RemoveEmpty<Something[Key]> extends never
    ? never
    : Key]: Something extends Record<string, unknown>
    ? ExcludeEmptyProperties<Something[Key]>
    : Exclude<Something[Key], '' | [] | undefined | null>;
};

type RemoveEmpty<Something> = Exclude<Something, '' | [] | undefined | null>;
```

Reply:
I have updated the generic naming but, as we agreed, the function typing will remain the same, as it got complicated to exclude null/undefined values from the return type.

Reply:
Yep, my bad, I was blinded by the pleasure of writing TS types